planning/scenario_selector/test/test_scenario_selector_node_interface.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ament_index_cpp/get_package_share_directory.hpp"
+#include "planning_interface_test_manager/planning_interface_test_manager.hpp"
+#include "planning_interface_test_manager/planning_interface_test_manager_utils.hpp"
+#include "scenario_selector/scenario_selector_node.hpp"
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <vector>
+
+TEST(PlanningModuleInterfaceTest, NodeTestWithExceptionTrajectoryLaneDrivingMode)
+{
+  rclcpp::init(0, nullptr);
+
+  auto test_manager = std::make_shared<planning_test_utils::PlanningInterfaceTestManager>();
+
+  auto node_options = rclcpp::NodeOptions{};
+  node_options.append_parameter_override("update_rate", 10.0);
+  node_options.append_parameter_override("th_max_message_delay_sec", INFINITY);
+  node_options.append_parameter_override("th_arrived_distance_m", 1.0);
+  node_options.append_parameter_override("th_stopped_time_sec", 1.0);
+  node_options.append_parameter_override("th_stopped_velocity_mps", 0.01);
+  auto test_target_node = std::make_shared<ScenarioSelectorNode>(node_options);
+
+  // publish necessary topics from test_manager
+  test_manager->publishTF(test_target_node, "/tf");
```

Comment:
just question, 

```
// Wait for first tf
while (rclcpp::ok()) {
....
}
```
is removed but `tf` is still needed?

Reply:
@kyoichi-sugahara If you created another PR [here](https://github.com/autowarefoundation/autoware.universe/pull/3414), then please replace the TF with current_pose and get rid of the TF-related code completely. Then we need to publish TF for this node anymore.

Reply:
@kosuke55 
hmm you are right.
I will change ego position source from `tf` to `odom` in different PR.
And remove tf publisher from test.

Reply:
@TakaHoribe 
Changed source from tf to odom in this [PR](https://github.com/autowarefoundation/autoware.universe/pull/3414)

Reply:
it seems deleted in https://github.com/autowarefoundation/autoware.universe/pull/3376/commits/0715db35655cc939f1f510f3e45261eaf64f752e but still exist :thinking:

Reply:
@kosuke55  cc: @TakaHoribe 
Thank you for checking!
I forgot there are 2 tests and I should have deleted from both test case.
I deleted from both.